sdk/core/core-http/test/policies/atomSerializationPolicyTests.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { assert } from "chai";
+import { HttpHeaders } from "../../lib/httpHeaders";
+import { HttpOperationResponse } from "../../lib/httpOperationResponse";
+import { HttpClient, AtomXmlOperationSpec, AtomXmlSerializer } from "../../lib/coreHttp";
+import { atomSerializationPolicy } from "../../lib/policies/atomSerializationPolicy";
+import { RequestPolicyOptions } from "../../lib/policies/requestPolicy";
+import { WebResource } from "../../lib/webResource";
+
+describe("atomSerializationPolicy", function() {
+  it("should throw an error if receiving a non-XML response body", async function() {
+    const request: WebResource = createRequest({
+      serializer: new MockSerializer()
+    });
+    const mockClient: HttpClient = {
+      sendRequest: (req) =>
+        Promise.resolve({
+          request: req,
+          status: 200,
+          headers: new HttpHeaders({ "Content-Type": "application/json" }),
+          bodyAsText: `{ "simple": "JSONobject" }`
+        })
+    };
+
+    const policy = atomSerializationPolicy().create(mockClient, new RequestPolicyOptions());
+    try {
+      await policy.sendRequest(request);
+      assert.deepEqual(true, false, "Error must be thrown");
+    } catch (err) {
+      assert.deepEqual(err.message, "ResponseNotInAtomXMLFormat");
+      assert.deepEqual(err.code, "PARSE_ERROR");
+    }
+  });
+
+  it("with xml response body, application/xml content-type and AtomXMLOperationSpec", async function() {
+    const request: WebResource = createRequest({
+      serializer: new MockSerializer()
+    });
+
+    const expectedResult = {
+      LockDuration: "PT2M",
+      MaxSizeInMegabytes: "1024",
+      _: {
+        ContentRootElement: "QueueDescription",
+        id:
+          "https://servicebuslocalperftestspremium1.servicebus.windows.net/testQueuePath4?api-version=2017-04",
+        title: "testQueuePath4",
+        author: { name: "servicebuslocalperftestspremium1" }
+      },
+      QueueName: "testQueuePath4"
+    };
+
+    const mockClient: HttpClient = {
+      sendRequest: (req) =>
+        Promise.resolve({
+          request: req,
+          status: 200,
+          headers: new HttpHeaders({
+            "content-type": "application/xml"
+          }),
+          bodyAsText: `<entry xmlns="http://www.w3.org/2005/Atom"><id>https://servicebuslocalperftestspremium1.servicebus.windows.net/testQueuePath4?api-version=2017-04</id><title type="text">testQueuePath4</title><author><name>servicebuslocalperftestspremium1</name></author><content type="application/xml"><QueueDescription xmlns="http://schemas.microsoft.com/netservices/2010/10/servicebus/connect" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><LockDuration>PT2M</LockDuration><MaxSizeInMegabytes>1024</MaxSizeInMegabytes></QueueDescription></content></entry>`,
```

Comment:
Just to expand on the comment in MockSerializer, right now `bodyAsText` can be an empty string and the test will still pass. That makes this test seem not very valuable. I'd love to see the test actually parse this body, and likewise a test that  accepts user unput as a JSON object and converts it into the proper xml document.

Reply:
There are tests on the consumer package which is where the Atom specific serialization is being done. So these tests here assume that the serialization code is plugged in properly and mostly unit tests the policy that is core to core-http.
Thus, it seems like the tests in Service Bus already provide the coverage we are looking for.

Also - conceptually, there isn't any true test for the serialization stuff right?... because it is specifically written to generate what the service expects and not all that generic. 

As for the xml and other utilities in core-http, looks like they already have some coverage in place. 
For the Atom XML specific utilities - not sure if I commented earlier, but there is something buggy about xml2js parser library, and the tests were failing because the settings were being overridden and seems like they are internally treated as static and not new for each instance of the parser.
Needed a workaround for this to make the feature work, more investigation/work may be needed to more cleanly fix the xml2js bug.
So maybe we can add more tests specific for the AtomXML utilities when the bug is fixed. Filed issue for it at - https://github.com/Azure/azure-sdk-for-js/issues/5344

Added some tests and updated MockSerializer to be a TestSerializer instead (mostly re-uses the Queue serializer from Service Bus for an example)

Reply:
The tests on the consumer will test the serialization and deserialization parts based on the spec that the service is using to serialize & deserialize

The tests in core-http should test the xml to json and json to xml conversions which should be agnostic to what the service expects